crates/ide_completion/src/completions/fn_param.rs

```diff
@@ -161,16 +155,16 @@ fn should_add_self_completions(ctx: &CompletionContext, param_list: &ast::ParamL
     inside_impl && no_params
 }
 
-fn comma_wrapper(ctx: &CompletionContext) -> Option<(impl Fn(&str) -> String, TextRange, String)> {
+fn comma_wrapper(ctx: &CompletionContext) -> Option<(impl Fn(&str) -> String, TextRange)> {
     let param = ctx.token.ancestors().find(|node| node.kind() == SyntaxKind::PARAM)?;
-
+    let is_mut_token = (ctx.token.kind() == SyntaxKind::MUT_KW).then(|| &ctx.token);
```

Comment:
I discovered this regression during testing: if current token is exactly `mut` kw, for example `fn foo(mut$0)`, the ancestor PARAM will become which include the R_PAREN and removes it incorrectly.
```
[crates/ide_completion/src/completions/fn_param.rs:160] &param = PARAM@27..31
  IDENT_PAT@27..31
    MUT_KW@27..30 "mut"
    ERROR@30..31
      R_PAREN@30..31 ")"
``` 
I decided just to special case it but not sure if there are more situations the ancestor PARAM range should be adjusted.

Reply:
Oh that is a bug in the parser I am pretty sure. Iirc we have the invariant that nodes have balanced parentheses which isn't the case here

Reply:
Should be fixed with https://github.com/rust-lang/rust-analyzer/pull/12090